src/main/kotlin/com/freya02/botcommands/internal/utils/ReflectionMetadata.kt

```diff
@@ -145,33 +109,37 @@ internal object ReflectionMetadata {
         scannedParams = true
     }
 
-    internal inline fun <reified A : Annotation> KParameter.hasAnnotation_(): Boolean {
-        return (paramMetadataMap[this]
-            ?: throwUser("Tried to access a KParameter which hasn't been scanned: $this, the parameter must be accessible and in the search path")).annotationMap[A::class] != null
-    }
-
-    internal inline fun <reified A : Annotation> KParameter.findAnnotation_(): A? {
-        return (paramMetadataMap[this]
-            ?: throwUser("Tried to access a KParameter which hasn't been scanned: $this, the parameter must be accessible and in the search path")).annotationMap[A::class] as? A
-    }
-
     internal val KParameter.isNullable: Boolean
-        get() = (paramMetadataMap[this]
-            ?: throwUser("Tried to access a KParameter which hasn't been scanned: $this, the parameter must be accessible and in the search path")).isNullable
-
-    internal val KParameter.function
-        get() = (paramMetadataMap[this]
-            ?: throwUser("Tried to access a KParameter which hasn't been scanned: $this, the parameter must be accessible and in the search path")).function
+        get() {
+            val metadata = functionMetadataMap[function.javaMethod]
+                ?: throwUser("Tried to access a Method which hasn't been scanned: $this, the method must be accessible and in the search path")
+            val isNullableAnnotated =
+                metadata.nullabilities[index - 1] // -1 because 0 is actually the instance parameter
+            val isNullableMarked = type.isMarkedNullable
+
+            return isNullableAnnotated || isNullableMarked
+        }
 
-    internal val KParameter.isJava
-        get() = (paramMetadataMap[this]
-            ?: throwUser("Tried to access a KParameter which hasn't been scanned: $this, the parameter must be accessible and in the search path")).isJava
+    private val handle: MethodHandle by lazy {
+        val kParameterImplClazz = Class.forName("kotlin.reflect.jvm.internal.KParameterImpl")
+        MethodHandles.publicLookup()
+            .unreflect(kParameterImplClazz.kotlin.memberProperties.find { it.name == "callable" }!!.javaGetter!!)
+    }
+    internal val KParameter.function: KFunction<*>
+        get() {
+            val callable = handle.invoke(this)
+            return callable as? KFunction<*>
+                ?: throwInternal("Unable to get the function of a KParameter, callable is: $callable")
+        }
```

Comment:
This could just be using a java method to use `KParameterImpl#getCallable`, as it is public (but internal)